### ISSUE TITLE
Discrepancy between AS collisions in RUI and EUI

### ISSUE THREAD

**Issue by danqaurooni** [hubmapconsortium/hra-ui#837]
There is a discrepancy between the AS tags in the RUI and the EUI for the JSON below:

[pryhuber.json](https://github.com/user-attachments/files/17906311/pryhuber.json)

When I open it in the RUI, I get collision with the following ASes (note that the ASes were added manually):
<img width="392" alt="Screenshot 2024-11-25 at 11 55 24 AM" src="https://github.com/user-attachments/assets/cd08e2b8-e111-4500-86f5-20bab238c579">

But when I open it up in the EUI (https://hubmapconsortium.github.io/hra-registrations/hubmap-lung-pryhuber-2023/) and filter the results for that block, I don't get the same collisions:

<img width="430" alt="Screenshot 2024-11-25 at 11 56 21 AM" src="https://github.com/user-attachments/assets/c25ebb87-1a58-4435-b90d-5ff552593f18">

**Comment by bherr2** [comment]
Ok, I replaced the template with the latest EUI version, but still don't see it in the tree. I see all 4 annotations in the RUI location in the API response where it says what AS were collided with: <https://apps.humanatlas.io/api/v1/ontology-term-occurences?token=57b0c5792985a62bd7c9d1562505effd>. 3 of the 4 annotations aren't in <https://apps.humanatlas.io/api/v1/ontology-tree-model>, so it's probably something in how the tree is constructed (the ontologies aren't trees, so this is a possibility). It seems to be a bug in hra-api and might take a bit to resolve.

**Comment by bherr2** [comment]
Ok and looking at this further, its even worse. 3 of those annotations are no longer in the lung ASCT+B table. The tree is constructed from the tables, so that is why it doesn't show up in the list.

**Comment by bherr2** [comment]
@emquardokus these 3 structures are tagged in the 3d reference organ for lung, but do not occur in the v1.4 lung table:

* http://purl.obolibrary.org/obo/UBERON_0002168 (left lung)
* http://purl.obolibrary.org/obo/UBERON_0008953 (lower lobe of left lung)
* http://purl.org/sig/ont/fma/fma7508 (Left posterior basal segmental bronchus)

**Comment by emquardokus** [comment]
> @emquardokus these 3 structures are tagged in the 3d reference organ for lung, but do not occur in the v1.4 lung table:
> 
>     * http://purl.obolibrary.org/obo/UBERON_0002168 (left lung)
> 
>     * http://purl.obolibrary.org/obo/UBERON_0008953 (lower lobe of left lung)
> 
>     * http://purl.org/sig/ont/fma/fma7508 (Left posterior basal segmental bronchus)

@bherr2 v1.0 lung had left and right lung, but it was decided to make it a generic lung, so no lefts and rights of anything are in the ASCT+B table only in the 3D ref. models.
Same for all the L/R laterality organs like kidney, ureter, mammary gland, ovary, fallopian tube etc---only in 3D ref models, not in ASCT+B tables.

**Comment by bherr2** [comment]
Yeah, so I'm wondering how we get those anatomical structure tags from the 3D reference organs when they use the RUI to show up in the EUI's AS partonomy on the left. The AS partonomy is built from the ASCT+B tables, so terms that don't show up there don't get highlighted even though there are RUI locations that have those terms.

**Comment by emquardokus** [comment]
hmmm, good point.  I don't think there is any intention on going back to laterality in ASCT+B tables due to high redundancy, which was why it was simplified.

**Comment by bherr2** [comment]
Ok, yeah, we'll have to think about how to handle this
